File: python_tutorial/test_geometry_sphere.py
import os,sys
import numpy as np

# test area on a sphere
if '../py_funs' not in sys.path:
   sys.path.append('../py_funs')

import geometry_sphere as GS

if 0:
   #test area on an ellipsoid calc (areaint.m)
   a              = 6378273   # semi-major axis
   ecc            = .1        # eccentricity of ellipse
   ellipsoid_mat  = [a,ecc]   # ellipsoid formatted ala matlab

   if 1:
      lons  = np.array([40,50,50,40])
      lats  = np.array([60,60,70,70])

      if 0:
         print('lons')
         print(lons)
         print(lons*np.pi/180.)
         #
         print('lats')
         print(lats)
         print(lats*np.pi/180.)

   area  = GS.area_polygon_ellipsoid(lons,lats,ellipsoid_mat=ellipsoid_mat)
   print('Area: '+str(area)+' m^2')

if 1:
   # test conformal lat calc + its inverse
   a              = 6378273   # semi-major axis
   ecc            = .2        # eccentricity of ellipse
   ellipsoid_mat  = [a,ecc]   # ellipsoid formatted ala matlab

   phi   = 70
   chi   = GS.convertlat_geodetic2conformal(ecc,phi,inverse=False,radians=False)
   phi2  = GS.convertlat_geodetic2conformal(ecc,chi,inverse=True,radians=False)
   print(phi,chi,phi2)
   

File: python_tutorial/matlab/test_geometry_sphere.m
format long

if 0
   a              = 6378273;  % semi-major axis (m)
   ecc            = .1;       % eccentricity of ellipse
   ellipsoid_mat  = [a,ecc];  % ellipsoid formatted ala matlab

   if 1
      lons  = [40,50,50,40];
      lats  = [60,60,70,70];
   end

   A  = areaint(lats,lons,ellipsoid_mat,'degrees')
else
   ecc   = .2;
   phi   = 70
   chi   = convertlat([1,ecc], phi, 'geodetic', 'conformal', 'degrees')
   phi2  = convertlat([1,ecc], chi, 'conformal','geodetic', 'degrees')
end
